File: apps/wx-server/models/order.py
from datetime import datetime
from sqlalchemy import Column, String, Integer, DateTime, Float, func
from .base import BaseModel


class Order(BaseModel):
    """订单信息实体"""
    __tablename__ = 'orders'
    
    id = Column(String(50), primary_key=True)  # 订单ID
    user_id = Column(String(50), nullable=False)  # 用户ID
    product_id = Column(String(50), nullable=False)  # 商品ID（课程ID或咨询预约ID）
    type = Column(Integer, nullable=False)  # 商品类型（1-课程，2-咨询）
    amount = Column(Float, default=0.0)  # 订单金额
    status = Column(Integer, default=0)  # 支付状态（0-未支付，1-已支付，2-已退款）

    def to_dict(self):
        """转换为字典"""
        return {
            'id': self.id,
            'user_id': self.user_id,
            'product_id': self.product_id,
            'type': self.type,
            'amount': self.amount,
            'status': self.status,
            'create_time': self.create_time.isoformat() if self.create_time else None,
            'update_time': self.update_time.isoformat() if self.update_time else None
        }

File: apps/server-spring/qiyun-ur/src/main/java/com/ruoyi/ur/domain/entity/Order.java
package com.ruoyi.ur.domain.entity;

import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;
import java.util.Date;

@Data
@ApiModel(value = "Order", description = "订单信息实体")
public class Order {
    @ApiModelProperty("订单ID")
    private String id;
    
    @ApiModelProperty("用户ID")
    private String userId;
    
    @ApiModelProperty("商品ID（课程ID或咨询预约ID）")
    private String productId;
    
    @ApiModelProperty("商品类型（1-课程，2-咨询）")
    private Integer type;
    
    @ApiModelProperty("订单金额")
    private Double amount;
    
    @ApiModelProperty("支付状态（0-未支付，1-已支付，2-已退款）")
    private Integer status;
    
    @ApiModelProperty("创建时间")
    private Date createTime;
    
    @ApiModelProperty("更新时间")
    private Date updateTime;
}